runtime/parser/expression.go

```diff
@@ -161,29 +172,36 @@ func defineExpr(def any) {
 		defineExpr(prefixExpr{
 			tokenType:    def.tokenType,
 			bindingPower: def.bindingPower,
-			nullDenotation: func(p *parser, right ast.Expression, tokenRange ast.Range) ast.Expression {
+			nullDenotation: func(p *parser, right ast.Expression, tokenRange ast.Range) (ast.Expression, error) {
 				return ast.NewUnaryExpression(
 					p.memoryGauge,
 					def.operation,
 					right,
 					tokenRange.StartPos,
-				)
+				), nil
 			},
 		})
 
 	case postfixExpr:
 		tokenType := def.tokenType
 		setExprLeftBindingPower(tokenType, def.bindingPower)
-		setExprLeftDenotation(
+		err = setExprLeftDenotation(
 			tokenType,
-			func(p *parser, token lexer.Token, left ast.Expression) ast.Expression {
+			func(p *parser, token lexer.Token, left ast.Expression) (ast.Expression, error) {
 				return def.leftDenotation(p, left, token.Range)
 			},
 		)
 
 	default:
 		panic(errors.NewUnreachableError())
 	}
+
+	// It is OK to panic here, since this method is only called inside `init()`.
+	// Returning the error from here explodes the error handling in `init` function.
+	// So panic it here, which has the same effect.
+	if err != nil {
+		panic(err)
```

Comment:
It's already refactored now, but we can keep the panics in the parser "setup" functions, like in `setExprLeftDenotation` – they're only called when the parser package is initialized, not during usage/parsing itself

Reply:
reverted in 30f9a812d5725ccbbda8a4ca0a4c823738d709af